com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/HDShadow.hlsl

```diff
@@ -67,13 +74,31 @@ float GetPunctualShadowClosestDistance(HDShadowContext shadowContext, SamplerSta
         sd.rot2 = shadowContext.shadowDatas[shadowDataIndex + CubeMapFaceID(-L)].rot2;
     }
 
-    return EvalShadow_SampleClosestDistance_Punctual(sd, _ShadowmapAtlas, sampl, positionWS, L, lightPositionWS);
+    if (sd.isInCachedAtlas > 0) // This is a scalar branch.
```

Comment:
why not:

 return EvalShadow_SampleClosestDistance_Punctual(sd, sd.isInCachedAtlas > 0 ? _ShadowmapAtlas : _CachedShadowmapAtlas, sampl, positionWS, L, lightPositionWS);

will this be worse?

Reply:
It doesn't compile :( That's what I had before, but unity compiler complains